operations-stdlib/src/commonMain/kotlin/string/compareToDate/CompareToDate.kt

```diff
@@ -0,0 +1,92 @@
+package string.compareToDate
+
+import operation.StandardLogicOperation
+import string.StringUnwrapStrategy
+import utils.asList
+
+object CompareToDate: StandardLogicOperation, StringUnwrapStrategy {
+
+    private const val YEAR_LENGTH = 4
+
+    private const val DELIMITER = "-"
+    private const val DATE_INDEX = 0
+    private const val COMPARING_DATE_INDEX = 1
+
+    private const val YEAR_INDEX = 0
+    private const val MONTH_INDEX = 1
+    private const val DAY_INDEX = 2
+
+    private const val JANUARY = 1
+    private const val FEBRUARY = 2
+    private const val MARCH = 3
+    private const val MAY = 5
+    private const val JULY = 7
+    private const val AUGUST = 8
+    private const val OCTOBER = 10
+    private const val DECEMBER = 12
+
+    private val MONTHS_WITH_31_DAYS = intArrayOf(JANUARY, MARCH, MAY, JULY, AUGUST, OCTOBER, DECEMBER)
+
+    private const val DAY_30_DAYS_VALUE = 30
+    private const val DAY_MAX_VALUE = 31
+    private const val DAY_FEBRUARY_LEAP_YEAR_MAX_VALUE = 29
+    private const val DAY_FEBRUARY_DEFAULT_MAX_VALUE = 28
+    private const val DAY_MIN_VALUE = 1
+    private const val MONTH_MIN_VALUE = 1
+    private const val MONTH_MAX_VALUE = 12
+
+    override fun evaluateLogic(expression: Any?, data: Any?): Any? =
+        expression.asList.toCompareDateParameters()?.invokeCompare()
+
+    private fun List<Any?>.toCompareDateParameters(): CompareDateArguments? = runCatching {
+        val baseDate = (get(DATE_INDEX) as String).splitDate()
+        val compareDate = (get(COMPARING_DATE_INDEX) as String).splitDate()
+
+        if (!baseDate.isContainsValidDate() || !compareDate.isContainsValidDate()) {
+            return null
+        }
+        val baseDateParameters = baseDate.toDateParameters() ?: return null
+        val compareDateParameters = compareDate.toDateParameters() ?: return null
+        CompareDateArguments(
+            baseDate = baseDateParameters,
+            compareDate = compareDateParameters
+        )
+    }.getOrNull()
+
+    private fun String.splitDate() = split(DELIMITER)
+
+    private fun Int.isFebruary() = this == FEBRUARY
+
+    private fun Int.contains31Days() = MONTHS_WITH_31_DAYS.contains(this)
+
+    private fun List<String>.toDateParameters(): DateParameters? = runCatching {
+        DateParameters(
+            year = get(YEAR_INDEX).toInt(),
+            month = get(MONTH_INDEX).toInt(),
+            day = get(DAY_INDEX).toInt()
+        )
+    }.getOrNull()
+
+    private fun isMonthContainsCorrectNumberOfDays(year: Int, month: Int, day: Int): Boolean = when {
+            month.isFebruary() && year.isLeapYear() && day <= DAY_FEBRUARY_LEAP_YEAR_MAX_VALUE -> true
+            month.isFebruary() && !year.isLeapYear() && day <= DAY_FEBRUARY_DEFAULT_MAX_VALUE -> true
+            month.contains31Days() && day <= DAY_MAX_VALUE -> true
+            !month.contains31Days() && !month.isFebruary() && day <= DAY_30_DAYS_VALUE -> true
+            else -> false
+    }
```

Comment:
When you add an enum class, it can be replaced with sth like that:
```
    private fun isMonthContainsCorrectNumberOfDays(year: Int, month: Int, day: Int): Boolean {
        return if (mapIntToMonth(month) == Month.FEBRUARY && year.isLeapYear) {
            day <= DAY_FEBRUARY_LEAP_YEAR_MAX_VALUE
        } else day <= mapIntToMonth(month).standardMaxDays
    }
```